opencti-platform/opencti-graphql/tests/02-integration/07-connector/importCsv-connector-test.ts

```diff
@@ -0,0 +1,59 @@
+import { describe, it, expect } from 'vitest';
+import { ADMIN_USER, testContext } from '../../utils/testQuery';
+import { processCSVforWorkers } from '../../../src/connector/importCsv/importCsv-connector';
+import { csvMapperMockSimpleCities } from './importCsv-connector/csv-mapper-cities';
+import { createWork, findById as findWorkById } from '../../../src/domain/work';
+import { IMPORT_CSV_CONNECTOR } from '../../../src/connector/importCsv/importCsv';
+import type { CsvMapperParsed } from '../../../src/modules/internal/csvMapper/csvMapper-types';
+import { resolveUserByIdFromCache } from '../../../src/domain/user';
+import type { AuthUser } from '../../../src/types/user';
+import conf from '../../../src/config/conf';
+import { IMPORT_STORAGE_PATH } from '../../../src/modules/internal/document/document-domain';
+import { fileToReadStream, uploadToStorage } from '../../../src/database/file-storage-helper';
+import type { CsvBundlerIngestionOpts } from '../../../src/parser/csv-bundler';
+
+describe('Verify internal importCsv connector', () => {
+  let work: any;
+
+  it('should import_csv_built_in_connector configuration be not changed on test', async () => {
+    // Small bulk size to validate that there is no regression when there is more data than bulk size.
+    expect(conf.get('import_csv_built_in_connector:bulk_creation_size'), 'Please be careful when changing bulk_creation_size in tests config').toBe(5);
+  });
+
+  it('should upload csv and create work that is use for this test', async () => {
+    const file = fileToReadStream('./tests/02-integration/07-connector/importCsv-connector', 'csv-file-cities.csv', 'csv-file-cities.csv', 'text/csv');
+    const uploadedFile = await uploadToStorage(testContext, ADMIN_USER, `${IMPORT_STORAGE_PATH}/global`, file, {});
+    expect(uploadedFile).toBeDefined();
+    expect(uploadedFile.upload.id).toBe('import/global/csv-file-cities.csv');
+
+    work = await createWork(testContext, ADMIN_USER, IMPORT_CSV_CONNECTOR, '[File] Import csv for test', 'sourceTest');
+  });
+
+  it('should convert csv lines to bundle when line count < bulk_creation_size', async () => {
+    const user = await resolveUserByIdFromCache(testContext, ADMIN_USER.id) as AuthUser;
+
+    const mapperOpts: CsvBundlerIngestionOpts = {
+      connectorId: 'test-connector',
+      applicantUser: user,
+      csvMapper: csvMapperMockSimpleCities as CsvMapperParsed,
+      entity: undefined,
+      workId: work.id
+    };
+    const totalObjectsCount = await processCSVforWorkers(testContext, 'import/global/csv-file-cities.csv', mapperOpts);
+
+    // Bulk size = 5
+    //
+    // 3 first city line => same city on 2 first lines: 2 city + 'label1', 1 city + label2  = 5 objects
+    // next 5 lines => 1 skip line, 4 cities, 1 label = 5 objects
+    // next 5 lines => 5 cities + 1 label = 6 objects
+    // next 5 lines => 3 cities + 1 label ; + same city + 1 label => 6 objects (in 2 bundles)
```

Comment:
Shouldn't it be only 1 bundle if the city is a complete duplicate?

Reply:
it's not a complete duplicate, it's same city but with different label.